cluster/gce/gci/configure-helper.sh

```diff
@@ -0,0 +1,684 @@
+# More specifically, it prepares dirs and files, sets the variable value
+# in the manifests, and copies them to /etc/kubernetes/manifests.
+function start-etcd-servers {
+  echo "Start etcd pods"
+  if [[ -d /etc/etcd ]]; then
+    rm -rf /etc/etcd
+  fi
+  if [[ -e /etc/default/etcd ]]; then
+    rm -f /etc/default/etcd
+  fi
+  if [[ -e /etc/systemd/system/etcd.service ]]; then
+    rm -f /etc/systemd/system/etcd.service
+  fi
+  if [[ -e /etc/init.d/etcd ]]; then
+    rm -f /etc/init.d/etcd
+  fi
+  prepare-log-file /var/log/etcd.log
+  prepare-etcd-manifest "" "4001" "2380" "200m" "etcd.manifest"
+
+  prepare-log-file /var/log/etcd-events.log
+  prepare-etcd-manifest "-events" "4002" "2381" "100m" "etcd-events.manifest"
+}
+
+# Calculates the following variables based on env variables, which will be used
+# by the manifests of several kube-master components.
+#   CLOUD_CONFIG_VOLUME
+#   CLOUD_CONFIG_MOUNT
+#   DOCKER_REGISTRY
+function compute-master-manifest-variables {
+  CLOUD_CONFIG_VOLUME=""
+  CLOUD_CONFIG_MOUNT=""
+  if [[ -n "${PROJECT_ID:-}" && -n "${TOKEN_URL:-}" && -n "${TOKEN_BODY:-}" && -n "${NODE_NETWORK:-}" ]]; then
+    CLOUD_CONFIG_VOLUME="{\"name\": \"cloudconfigmount\",\"hostPath\": {\"path\": \"/etc/gce.conf\"}},"
+    CLOUD_CONFIG_MOUNT="{\"name\": \"cloudconfigmount\",\"mountPath\": \"/etc/gce.conf\", \"readOnly\": true},"
+  fi
+  DOCKER_REGISTRY="gcr.io/google_containers"
+  if [[ -n "${KUBE_DOCKER_REGISTRY:-}" ]]; then
+    DOCKER_REGISTRY="${KUBE_DOCKER_REGISTRY}"
+  fi
+}
+
+# A helper function for removing salt configuration and comments from a file.
+# This is mainly for preparing a manifest file.
+# $1: Full path of the file to manipulate
+function remove-salt-config-comments {
+  # Remove salt configuration
+  sed -i "/^[ |\t]*{[#|%]/d" $1
+  # Remove comments
+  sed -i "/^[ |\t]*#/d" $1
+}
+
+# Starts kubernetes apiserver.
+# It prepares the log file, loads the docker image, calculates variables, sets them
+# in the manifest file, and then copies the manifest file to /etc/kubernetes/manifests.
+#
+# Assumed vars (which are calculated in function compute-master-manifest-variables)
+#   CLOUD_CONFIG_VOLUME
+#   CLOUD_CONFIG_MOUNT
+#   DOCKER_REGISTRY
+function start-kube-apiserver {
+  echo "Start kubernetes api-server"
+  prepare-log-file /var/log/kube-apiserver.log
+
+  # Calculate variables and assemble the command line.
+  local params="--cloud-provider=gce --address=127.0.0.1 --etcd-servers=http://127.0.0.1:4001 --tls-cert-file=/etc/srv/kubernetes/server.cert --tls-private-key-file=/etc/srv/kubernetes/server.key --secure-port=443 --client-ca-file=/etc/srv/kubernetes/ca.crt --token-auth-file=/etc/srv/kubernetes/known_tokens.csv --basic-auth-file=/etc/srv/kubernetes/basic_auth.csv --allow-privileged=true --authorization-mode=ABAC --authorization-policy-file=/etc/srv/kubernetes/abac-authz-policy.jsonl --etcd-servers-overrides=/events#http://127.0.0.1:4002 ${APISERVER_TEST_ARGS:-}"
```

Comment:
Can you format this as one flag / line so that it's easier to read and maintain?